BAEL-223 - Integrate DynamoDB into a Spring Boot Application using Sp…

anataliocs opened the pull request:
Integrating DynamoDB Into a Spring Boot Application Using Spring Data

For Issue:  http://jira.baeldung.com/browse/BAEL-223

http://inprogress.baeldung.com/?p=18138&preview=true

pkhanal commented on spring-boot-data-dynamodb/src/main/java/com/baeldung/spring/data/es/config/DynamoDBConfig.java:
```diff
@@ -0,0 +1,41 @@
+        AmazonDynamoDB amazonDynamoDB = new AmazonDynamoDBClient(amazonAWSCredentials());
+        if (StringUtils.isEmpty(amazonDynamoDBEndpoint)) {
+            amazonDynamoDB.setEndpoint(amazonDynamoDBEndpoint);
+        }
```
Should be **StringUtils.isNotEmpty**
It does not make sense to set the endpoint when it is empty :)
You might got confused when you referred to the sample from https://github.com/derjust/spring-data-dynamodb

anataliocs replied:
This has been updated.  The org.springframework.util.StringUtils does not have a isNotEmpty method so I just changed it to NOT isEmpty

pkhanal commented on spring-boot-data-dynamodb/src/test/java/com/baeldung/spring/data/es/repository/ProductInfoRepositoryIntegrationTest.java:
```diff
@@ -0,0 +1,84 @@
+
+    @Before
+    @Ignore
+    public void setUp() throws Exception {
```
You have ```@BeforeClass``` in the article, however you have ```@Before``` here. Please make it consistent. Based on the feedback, I think it should be ```@BeforeClass```. And the method needs to be ```static``` when we use ```@BeforeClass```

anataliocs replied:
Updated

pkhanal commented on spring-boot-data-dynamodb/README.MD:
```diff
@@ -0,0 +1,2 @@
+###The Course
+The "REST With Spring" Classes: http://bit.ly/restwithspring
```
Why do we have this link in the README? May be it is not relevant here.

anataliocs replied:
updated @pkhanal

pkhanal commented on spring-boot-data-dynamodb/src/main/resources/application.properties:
```diff
@@ -0,0 +1,31 @@
+
+##jolokia.config.debug=true
+##endpoints.jolokia.enabled=true
+##endpoints.jolokia.path=jolokia
```
Let's cleanup these unused and commented out properties

anataliocs replied:
updated @pkhanal

pkhanal commented on spring-boot-data-dynamodb/src/main/resources/logback.xml:
```diff
@@ -0,0 +1,14 @@
+		<appender-ref ref="STDOUT" />
+	</root>
+
+</configuration>
```
We have indentation issue here. Check out how it is done is other projects. Usually **4 spaces** preferred over **tab**. It's a minor issue but it is good to consistent with other projects.

anataliocs replied:
![image](https://cloud.githubusercontent.com/assets/4750426/19377367/a6e49b10-919a-11e6-99ff-b3f2cb40a254.png)

It is 4 spaces in the actual code if you open it up.  It just is formatted to look like a tab in the github UI

anataliocs replied:
![image](https://cloud.githubusercontent.com/assets/4750426/19377388/d4355898-919a-11e6-9475-4d329ee28755.png)

It is 4 spaces.  The github UI makes it look like a tab.  @pkhanal

anataliocs replied:
Also @pkhanal this project was copied from an existing accepted project and was not changed

pkhanal commented on spring-boot-data-dynamodb/src/main/resources/templates/index.html:
```diff
@@ -0,0 +1,19 @@
+<script src="/webjars/bootstrap/3.3.4/js/bootstrap.min.js"></script>
+
+</body>
+</html>
```
Similar comment regarding indentation issue.

anataliocs replied:
![image](https://cloud.githubusercontent.com/assets/4750426/19377401/e7eb9b7c-919a-11e6-96e2-1c53b0a4012c.png)

It is 4 spaces. The github UI makes it look like a tab. @pkhanal

anataliocs replied:
Also this project was copied from an existing accepted project and was not changed @pkhanal

anataliocs commented:
@pkhanal I have updated the PR as per your feedback

pkhanal commented on spring-boot-data-dynamodb/src/test/java/com/baeldung/spring/data/es/repository/ProductInfoRepositoryIntegrationTest.java:
```diff
@@ -0,0 +1,84 @@
+
+    @BeforeClass
+    @Ignore
+    public static void setUp() throws Exception {
```
rename to **setup**

pkhanal commented on spring-boot-data-dynamodb/src/test/java/com/baeldung/spring/data/es/repository/ProductInfoRepositoryIntegrationTest.java:
```diff
@@ -0,0 +1,84 @@
+    public static void setUp() throws Exception {
+
+        try {
+            dynamoDBMapper = new DynamoDBMapper(amazonDynamoDB);
```
I am afraid you cannot access non static field inside the *static* method. Please fix the compilation error.

pkhanal commented on spring-boot-data-dynamodb/src/test/java/com/baeldung/spring/data/es/repository/ProductInfoRepositoryIntegrationTest.java:
```diff
@@ -0,0 +1,84 @@
+    private static final String EXPECTED_COST = "20";
+    private static final String EXPECTED_PRICE = "50";
+
+    @BeforeClass
```
You will need to import the *BeforeClass*

pkhanal commented on spring-boot-data-dynamodb/pom.xml:
```diff
@@ -0,0 +1,185 @@
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<groupId>com.baeldung</groupId>
+	<artifactId>spring-boot</artifactId>
```
Change the artifactId to **spring-boot-dynamodb**.

anataliocs commented:
Updated all requests and also cleaned up code and updated dependencies

Integration test runs successfully
![image](https://cloud.githubusercontent.com/assets/4750426/19614920/6453a678-97ae-11e6-97eb-7d494dbd1801.png)

mvn clean install runs successfully
![image](https://cloud.githubusercontent.com/assets/4750426/19614931/86a22ab0-97ae-11e6-84de-5b6578042f8f.png)
